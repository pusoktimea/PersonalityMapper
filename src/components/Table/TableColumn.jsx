import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const TableColumn = ({cellClassName, headerClassName, isHeader, children}) => {
  const baseClass = 'persmap-table_row_column';

  if (isHeader) {
    return (
      <th className={cx(
        'persmap-table_header_item',
        headerClassName
      )}>
        {children}
      </th>
    );
  }
  return (
    <td className={cx(
      baseClass,
      cellClassName
    )}>
      {children}
    </td>
  );
};

TableColumn.propTypes = {
  cellClassName: PropTypes.string,
  headerClassName: PropTypes.string,
  isHeader: PropTypes.bool,
  children: PropTypes.node
};

TableColumn.defaultProps = {
  isHeader: false
};

export default TableColumn;
