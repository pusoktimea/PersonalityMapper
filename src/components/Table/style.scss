@import "../../styles/_colors.scss";

$baseClass: '.activate-table';

#{$baseClass} {
  border-collapse: collapse;
  width: 100%;
  &_body {
    #{$baseClass}_row {
      border-top: 1px solid $alto;
      border-bottom: 1px solid $alto;

      &:last-child {
        border-bottom: none;
      }

      &--selected {
        background: $jungleGreen !important;
        color: $white;

        &:hover {
          background: $jungleGreen !important;
        }
      }

      &_column {
        padding: 8px;
        line-height: 1.42857143;
      }
    }
  }

  &_header_item{
    padding: 8px;
    line-height: 1.42857143;
  }

  &--bordered {
    border: 1px solid $alto;
    border-collapse: collapse;

    #{$baseClass} {
      &_header_item, &_row_column {
        border: 1px solid $alto;
      }
    }
  }

  &--hovered {
    #{$baseClass}_body {
      #{$baseClass}_row:hover {
        background: rgba(0, 0, 0, 0.1);
      }
    }
  }

  &--striped {
    #{$baseClass}_body {
      #{$baseClass}_row--odd {
        background: $gallery;

        &:hover {
          background: rgba(0, 0, 0, 0.1);
        }
      }
    }
  }

  &--condensed {
    #{$baseClass}_body {
      #{$baseClass}_row_column {
        padding: 5px;
      }
    }
  }

  &--selectable {
    #{$baseClass}_row {
      cursor: pointer;
    }
  }
}