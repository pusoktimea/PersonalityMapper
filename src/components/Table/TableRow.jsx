import React, {Component, Fragment} from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import Icon from '../Icon';
import TableColumn from './TableColumn';

class TableRow extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isSubDataVisible: false
    };
  }

  toggleSubData = (event) => {
    this.setState({
      isSubDataVisible: !this.state.isSubDataVisible
    }, () => {
      if (typeof this.props.onToggleSubData === 'function') {
        this.props.onToggleSubData();
      }
    });
    event.stopPropagation();
  }

  render() {
    const {row, rowClassName, isHeaderRow, selected, onSelect, onMultiSelect} = this.props;
    const {isSubDataVisible} = this.state;

    const baseClass = 'activate-table_row';
    const renderableColumns = Object.entries(row).filter(([key, _]) => (
      !['hasSubData'].includes(key)
    )).map(([_, value]) => value);

    return (
      <Fragment>
        <tr
          className={cx(
            baseClass,
            selected && `${baseClass}--selected`,
            rowClassName
          )}
          onClick={(event) => {
            if (event.metaKey) {
              typeof onMultiSelect === 'function' && onMultiSelect(row);
            } else {
              typeof onSelect === 'function' && onSelect(row);
            }
          }}
        >
          <TableColumn
            isHeader={isHeaderRow}
          >
            {
              isHeaderRow ?
                '' :
                row.hasSubData ?
                  <Icon
                    icon={isSubDataVisible ? 'chevron-up' : 'chevron-down'}
                    onClick={this.toggleSubData}
                    className={`${baseClass}--control`}
                  /> :
                  <Icon icon="plus-circle" />
            }
          </TableColumn>
          {renderableColumns.map((content, index) => (
            <TableColumn
              key={`activate-table-cell-${index}`}
              isHeader={isHeaderRow}
            >
              {content}
            </TableColumn>
          ))}
        </tr>
        {
          row.hasSubData && isSubDataVisible &&
          <tr>
            <td colSpan={renderableColumns.length + 1}>subdata content</td>
          </tr>
        }
      </Fragment>
    );
  }
}

TableRow.propTypes = {
  row: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.array
  ]).isRequired,
  rowClassName: PropTypes.string,
  isHeaderRow: PropTypes.bool,
  selected: PropTypes.bool,
  onSelect: PropTypes.func,
  onMultiSelect: PropTypes.func,
  onToggleSubData: PropTypes.func
};

TableRow.defaultProps = {
  isHeaderRow: false,
  selected: false
};

export default TableRow;
