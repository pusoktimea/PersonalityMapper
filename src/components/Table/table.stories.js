import React from 'react';
import {setAddon, storiesOf} from '@storybook/react';
import {withKnobs, boolean} from '@storybook/addon-knobs';
import JSXAddon from 'storybook-addon-jsx';

import Table from './index';

const items = [
  {
    id: '1',
    col1: '11',
    col2: '12',
    col3: '13'
  }, {
    id: '2',
    col1: '21',
    col2: '22',
    col3: '23',
    hasSubData: true
  }, {
    id: '3',
    col1: '31',
    col2: '32',
    col3: '33'
  }
];
const headers = ['id', 'col1', 'col2', 'col3'];

setAddon(JSXAddon);
const stories = storiesOf('Components', module);

stories.addDecorator(withKnobs);
stories.addWithJSX('Table', () => (
  <Table
    condensed={boolean('Condensed', true)}
    striped={boolean('Striped', true)}
    hovered={boolean('Hovered', true)}
    bordered={boolean('Bordered', true)}
    items={items}
    headers={headers}
  />
));
