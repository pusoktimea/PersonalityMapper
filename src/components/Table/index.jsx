import React, {Component} from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import TableRow from './TableRow';

import './style.scss';

class Table extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedRows: []
    };
  }

  selectRow = (row) => {
    this.setState({
      selectedRows: [row.id]
    }, () => {
      if (typeof this.props.onRowSelect === 'function') {
        this.props.onRowSelect();
      }
    });
  }

  multiSelectRow = (row) => {
    if (!this.state.selectedRows.includes(row.id)) {
      this.setState({
        selectedRows: [
          ...this.state.selectedRows,
          row.id
        ]
      }, () => {
        if (typeof this.props.onMultiRowSelect === 'function') {
          this.props.onMultiRowSelect();
        }
      });
    } else {
      this.setState({
        selectedRows: this.state.selectedRows.filter((item) => item !== row.id)
      }, () => {
        if (typeof this.props.onMultiRowSelect === 'function') {
          this.props.onMultiRowSelect();
        }
      });
    }
  }

  renderHeader = () => {
    return (
      <thead className="persmap-table_header">
        <TableRow
          row={this.props.headers}
          isHeaderRow
        />
      </thead>
    );
  }

  renderRows = () => {
    return (
      <tbody className="persmap-table_body">
        {this.props.items.map((row, index) => (
          <TableRow
            row={row}
            key={`persmap-table-row-${index}`}
            onSelect={this.selectRow}
            onMultiSelect={this.multiSelectRow}
            selected={this.state.selectedRows.includes(row.id)}
            rowClassName={(index + 1) % 2 !== 0 ? 'persmap-table_row--odd' : ''}
            onToggleSubData={this.props.onToggleSubData}
          />
        ))}
      </tbody>
    );
  }
  render() {
    const baseClass = 'persmap-table';

    const {
      className,
      bordered,
      hovered,
      condensed,
      striped,
      headers
    } = this.props;

    return (
      <table className={cx(
        baseClass,
        condensed && `${baseClass}--condensed`,
        bordered && `${baseClass}--bordered`,
        striped && `${baseClass}--striped`,
        hovered && `${baseClass}--hovered`,
        className
      )}>
        {headers && this.renderHeader()}
        {this.renderRows()}
      </table>
    );
  }
}

Table.propTypes = {
  items: PropTypes.array.isRequired,
  headers: PropTypes.array,
  className: PropTypes.string,
  bordered: PropTypes.bool,
  hovered: PropTypes.bool,
  condensed: PropTypes.bool,
  striped: PropTypes.bool,
  onRowSelect: PropTypes.func,
  onMultiRowSelect: PropTypes.func,
  onToggleSubData: PropTypes.func
};

Table.defaultProps = {
  bordered: true,
  hovered: true,
  condensed: true,
  striped: true
};

export default Table;
