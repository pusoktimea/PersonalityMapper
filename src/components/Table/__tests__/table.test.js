import 'raf/polyfill';

import React from 'react';
import {findDOMNode} from 'react-dom';
import {renderIntoDocument, Simulate} from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

import Table from '../index';

const testTableRows = [
  {
    id: '1',
    col1: '11',
    col2: '12',
    col3: '13'
  }, {
    id: '2',
    col1: '21',
    col2: '22',
    col3: '23',
    hasSubData: true
  }, {
    id: '3',
    col1: '31',
    col2: '32',
    col3: '33'
  }
];
const testTableHeaders = ['id', 'col1', 'col2', 'col3'];

let tableElement = null;
const onRowSelect = jest.fn();
const onMultiRowSelect = jest.fn();
const onToggleSubData = jest.fn();

describe('Table component test', () => {
  describe('Snapshot tests for each modifier', () => {
    it('renders table with all modifiers turned off and matches snapshot', () => {
      expect(
        renderer.create(
          <Table
            items={testTableRows}
            headers={testTableHeaders}
            condensed={false}
            bordered={false}
            hovered={false}
            striped={false}
          />
        ).toJSON()
      ).toMatchSnapshot();
    });

    it('renders table only with condensed turned on and matches snapshot', () => {
      expect(
        renderer.create(
          <Table
            items={testTableRows}
            headers={testTableHeaders}
            bordered={false}
            hovered={false}
            striped={false}
          />
        ).toJSON()
      ).toMatchSnapshot();
    });

    it('renders table only with bordered turned on and matches snapshot', () => {
      expect(
        renderer.create(
          <Table
            items={testTableRows}
            headers={testTableHeaders}
            condensed={false}
            hovered={false}
            striped={false}
          />
        ).toJSON()
      ).toMatchSnapshot();
    });

    it('renders table only with hovered turned on and matches snapshot', () => {
      expect(
        renderer.create(
          <Table
            items={testTableRows}
            headers={testTableHeaders}
            bordered={false}
            condensed={false}
            striped={false}
          />
        ).toJSON()
      ).toMatchSnapshot();
    });

    it('renders table only with striped turned on and matches snapshot', () => {
      expect(
        renderer.create(
          <Table
            items={testTableRows}
            headers={testTableHeaders}
            bordered={false}
            hovered={false}
            condensed={false}
          />
        ).toJSON()
      ).toMatchSnapshot();
    });

    it('renders table with all modifiers turned on and matches snapshot', () => {
      expect(
        renderer.create(
          <Table
            items={testTableRows}
            headers={testTableHeaders}
          />
        ).toJSON()
      ).toMatchSnapshot();
    });
  });

  describe('Event handlers testing', () => {
    it('renders into DOM', () => {
      tableElement = findDOMNode(renderIntoDocument(
        <Table
          items={testTableRows}
          headers={testTableHeaders}
          onRowSelect={onRowSelect}
          onMultiRowSelect={onMultiRowSelect}
          onToggleSubData={onToggleSubData}
        />
      ));

      expect(tableElement).toBeTruthy();
    });

    it(`renders table header with ${testTableHeaders.length + 1} cells`, () => {
      const tableHeader = tableElement.querySelector('.activate-table_header');

      expect(tableHeader).toBeTruthy();

      const headerCells = tableHeader.querySelectorAll('.activate-table_header_item');

      expect(headerCells).toBeTruthy();
      expect(headerCells.length).toBe(testTableHeaders.length + 1);
    });

    it(`renders ${testTableRows.length} rows with ${testTableHeaders.length + 1} cells`, () => {
      const tableRows = tableElement.querySelectorAll('.activate-table_body .activate-table_row');

      expect(tableRows).toBeTruthy();
      expect(tableRows.length).toBe(testTableRows.length);

      const rowCells = tableRows[0].querySelectorAll('.activate-table_row_column');

      expect(rowCells).toBeTruthy();
      expect(rowCells.length).toBe(testTableHeaders.length + 1);
    });

    it('calls onRowSelect when clicking a row', () => {
      const tableRows = tableElement.querySelectorAll('.activate-table_body .activate-table_row');

      expect(tableRows).toBeTruthy();

      Simulate.click(tableRows[0]);
      expect(onRowSelect).toBeCalled();
    });

    it('calls onToggleSubData when clicking subdata toggle', () => {
      const tableRows = tableElement.querySelectorAll('.activate-table_body .activate-table_row');

      expect(tableRows).toBeTruthy();

      const rowControl = tableRows[1].querySelector('.activate-table_row--control');

      expect(rowControl).toBeTruthy();

      Simulate.click(rowControl);
      expect(onToggleSubData).toBeCalled();
    });
  });
});
