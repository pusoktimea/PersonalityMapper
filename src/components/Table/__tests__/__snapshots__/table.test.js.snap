// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component test Snapshot tests for each modifier renders table only with bordered turned on and matches snapshot 1`] = `
<table
  className="activate-table activate-table--bordered"
>
  <thead
    className="activate-table_header"
  >
    <tr
      className="activate-table_row"
      onClick={[Function]}
    >
      <th
        className="activate-table_header_item"
      >
        
      </th>
      <th
        className="activate-table_header_item"
      >
        id
      </th>
      <th
        className="activate-table_header_item"
      >
        col1
      </th>
      <th
        className="activate-table_header_item"
      >
        col2
      </th>
      <th
        className="activate-table_header_item"
      >
        col3
      </th>
    </tr>
  </thead>
  <tbody
    className="activate-table_body"
  >
    <tr
      className="activate-table_row activate-table_row--odd"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-plus-circle"
          id={undefined}
          onClick={undefined}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        1
      </td>
      <td
        className="activate-table_row_column"
      >
        11
      </td>
      <td
        className="activate-table_row_column"
      >
        12
      </td>
      <td
        className="activate-table_row_column"
      >
        13
      </td>
    </tr>
    <tr
      className="activate-table_row"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-chevron-down activate-table_row--control"
          id={undefined}
          onClick={[Function]}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        2
      </td>
      <td
        className="activate-table_row_column"
      >
        21
      </td>
      <td
        className="activate-table_row_column"
      >
        22
      </td>
      <td
        className="activate-table_row_column"
      >
        23
      </td>
    </tr>
    <tr
      className="activate-table_row activate-table_row--odd"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-plus-circle"
          id={undefined}
          onClick={undefined}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        3
      </td>
      <td
        className="activate-table_row_column"
      >
        31
      </td>
      <td
        className="activate-table_row_column"
      >
        32
      </td>
      <td
        className="activate-table_row_column"
      >
        33
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table component test Snapshot tests for each modifier renders table only with condensed turned on and matches snapshot 1`] = `
<table
  className="activate-table activate-table--condensed"
>
  <thead
    className="activate-table_header"
  >
    <tr
      className="activate-table_row"
      onClick={[Function]}
    >
      <th
        className="activate-table_header_item"
      >
        
      </th>
      <th
        className="activate-table_header_item"
      >
        id
      </th>
      <th
        className="activate-table_header_item"
      >
        col1
      </th>
      <th
        className="activate-table_header_item"
      >
        col2
      </th>
      <th
        className="activate-table_header_item"
      >
        col3
      </th>
    </tr>
  </thead>
  <tbody
    className="activate-table_body"
  >
    <tr
      className="activate-table_row activate-table_row--odd"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-plus-circle"
          id={undefined}
          onClick={undefined}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        1
      </td>
      <td
        className="activate-table_row_column"
      >
        11
      </td>
      <td
        className="activate-table_row_column"
      >
        12
      </td>
      <td
        className="activate-table_row_column"
      >
        13
      </td>
    </tr>
    <tr
      className="activate-table_row"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-chevron-down activate-table_row--control"
          id={undefined}
          onClick={[Function]}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        2
      </td>
      <td
        className="activate-table_row_column"
      >
        21
      </td>
      <td
        className="activate-table_row_column"
      >
        22
      </td>
      <td
        className="activate-table_row_column"
      >
        23
      </td>
    </tr>
    <tr
      className="activate-table_row activate-table_row--odd"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-plus-circle"
          id={undefined}
          onClick={undefined}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        3
      </td>
      <td
        className="activate-table_row_column"
      >
        31
      </td>
      <td
        className="activate-table_row_column"
      >
        32
      </td>
      <td
        className="activate-table_row_column"
      >
        33
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table component test Snapshot tests for each modifier renders table only with hovered turned on and matches snapshot 1`] = `
<table
  className="activate-table activate-table--hovered"
>
  <thead
    className="activate-table_header"
  >
    <tr
      className="activate-table_row"
      onClick={[Function]}
    >
      <th
        className="activate-table_header_item"
      >
        
      </th>
      <th
        className="activate-table_header_item"
      >
        id
      </th>
      <th
        className="activate-table_header_item"
      >
        col1
      </th>
      <th
        className="activate-table_header_item"
      >
        col2
      </th>
      <th
        className="activate-table_header_item"
      >
        col3
      </th>
    </tr>
  </thead>
  <tbody
    className="activate-table_body"
  >
    <tr
      className="activate-table_row activate-table_row--odd"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-plus-circle"
          id={undefined}
          onClick={undefined}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        1
      </td>
      <td
        className="activate-table_row_column"
      >
        11
      </td>
      <td
        className="activate-table_row_column"
      >
        12
      </td>
      <td
        className="activate-table_row_column"
      >
        13
      </td>
    </tr>
    <tr
      className="activate-table_row"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-chevron-down activate-table_row--control"
          id={undefined}
          onClick={[Function]}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        2
      </td>
      <td
        className="activate-table_row_column"
      >
        21
      </td>
      <td
        className="activate-table_row_column"
      >
        22
      </td>
      <td
        className="activate-table_row_column"
      >
        23
      </td>
    </tr>
    <tr
      className="activate-table_row activate-table_row--odd"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-plus-circle"
          id={undefined}
          onClick={undefined}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        3
      </td>
      <td
        className="activate-table_row_column"
      >
        31
      </td>
      <td
        className="activate-table_row_column"
      >
        32
      </td>
      <td
        className="activate-table_row_column"
      >
        33
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table component test Snapshot tests for each modifier renders table only with striped turned on and matches snapshot 1`] = `
<table
  className="activate-table activate-table--striped"
>
  <thead
    className="activate-table_header"
  >
    <tr
      className="activate-table_row"
      onClick={[Function]}
    >
      <th
        className="activate-table_header_item"
      >
        
      </th>
      <th
        className="activate-table_header_item"
      >
        id
      </th>
      <th
        className="activate-table_header_item"
      >
        col1
      </th>
      <th
        className="activate-table_header_item"
      >
        col2
      </th>
      <th
        className="activate-table_header_item"
      >
        col3
      </th>
    </tr>
  </thead>
  <tbody
    className="activate-table_body"
  >
    <tr
      className="activate-table_row activate-table_row--odd"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-plus-circle"
          id={undefined}
          onClick={undefined}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        1
      </td>
      <td
        className="activate-table_row_column"
      >
        11
      </td>
      <td
        className="activate-table_row_column"
      >
        12
      </td>
      <td
        className="activate-table_row_column"
      >
        13
      </td>
    </tr>
    <tr
      className="activate-table_row"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-chevron-down activate-table_row--control"
          id={undefined}
          onClick={[Function]}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        2
      </td>
      <td
        className="activate-table_row_column"
      >
        21
      </td>
      <td
        className="activate-table_row_column"
      >
        22
      </td>
      <td
        className="activate-table_row_column"
      >
        23
      </td>
    </tr>
    <tr
      className="activate-table_row activate-table_row--odd"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-plus-circle"
          id={undefined}
          onClick={undefined}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        3
      </td>
      <td
        className="activate-table_row_column"
      >
        31
      </td>
      <td
        className="activate-table_row_column"
      >
        32
      </td>
      <td
        className="activate-table_row_column"
      >
        33
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table component test Snapshot tests for each modifier renders table with all modifiers turned off and matches snapshot 1`] = `
<table
  className="activate-table"
>
  <thead
    className="activate-table_header"
  >
    <tr
      className="activate-table_row"
      onClick={[Function]}
    >
      <th
        className="activate-table_header_item"
      >
        
      </th>
      <th
        className="activate-table_header_item"
      >
        id
      </th>
      <th
        className="activate-table_header_item"
      >
        col1
      </th>
      <th
        className="activate-table_header_item"
      >
        col2
      </th>
      <th
        className="activate-table_header_item"
      >
        col3
      </th>
    </tr>
  </thead>
  <tbody
    className="activate-table_body"
  >
    <tr
      className="activate-table_row activate-table_row--odd"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-plus-circle"
          id={undefined}
          onClick={undefined}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        1
      </td>
      <td
        className="activate-table_row_column"
      >
        11
      </td>
      <td
        className="activate-table_row_column"
      >
        12
      </td>
      <td
        className="activate-table_row_column"
      >
        13
      </td>
    </tr>
    <tr
      className="activate-table_row"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-chevron-down activate-table_row--control"
          id={undefined}
          onClick={[Function]}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        2
      </td>
      <td
        className="activate-table_row_column"
      >
        21
      </td>
      <td
        className="activate-table_row_column"
      >
        22
      </td>
      <td
        className="activate-table_row_column"
      >
        23
      </td>
    </tr>
    <tr
      className="activate-table_row activate-table_row--odd"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-plus-circle"
          id={undefined}
          onClick={undefined}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        3
      </td>
      <td
        className="activate-table_row_column"
      >
        31
      </td>
      <td
        className="activate-table_row_column"
      >
        32
      </td>
      <td
        className="activate-table_row_column"
      >
        33
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table component test Snapshot tests for each modifier renders table with all modifiers turned on and matches snapshot 1`] = `
<table
  className="activate-table activate-table--condensed activate-table--bordered activate-table--striped activate-table--hovered"
>
  <thead
    className="activate-table_header"
  >
    <tr
      className="activate-table_row"
      onClick={[Function]}
    >
      <th
        className="activate-table_header_item"
      >
        
      </th>
      <th
        className="activate-table_header_item"
      >
        id
      </th>
      <th
        className="activate-table_header_item"
      >
        col1
      </th>
      <th
        className="activate-table_header_item"
      >
        col2
      </th>
      <th
        className="activate-table_header_item"
      >
        col3
      </th>
    </tr>
  </thead>
  <tbody
    className="activate-table_body"
  >
    <tr
      className="activate-table_row activate-table_row--odd"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-plus-circle"
          id={undefined}
          onClick={undefined}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        1
      </td>
      <td
        className="activate-table_row_column"
      >
        11
      </td>
      <td
        className="activate-table_row_column"
      >
        12
      </td>
      <td
        className="activate-table_row_column"
      >
        13
      </td>
    </tr>
    <tr
      className="activate-table_row"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-chevron-down activate-table_row--control"
          id={undefined}
          onClick={[Function]}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        2
      </td>
      <td
        className="activate-table_row_column"
      >
        21
      </td>
      <td
        className="activate-table_row_column"
      >
        22
      </td>
      <td
        className="activate-table_row_column"
      >
        23
      </td>
    </tr>
    <tr
      className="activate-table_row activate-table_row--odd"
      onClick={[Function]}
    >
      <td
        className="activate-table_row_column"
      >
        <span
          className="activate-icon fa fa-plus-circle"
          id={undefined}
          onClick={undefined}
        />
      </td>
      <td
        className="activate-table_row_column"
      >
        3
      </td>
      <td
        className="activate-table_row_column"
      >
        31
      </td>
      <td
        className="activate-table_row_column"
      >
        32
      </td>
      <td
        className="activate-table_row_column"
      >
        33
      </td>
    </tr>
  </tbody>
</table>
`;
